### PR TITLE
#4 Issue fix

### PR DESCRIPTION
**Pull request by adambarath**
Fix for Issue https://github.com/laurentiustamate94/netstandard-storage/issues/4

### CODE REVIEW

**Comment by adambarath** [comment]
@laurentiustamate94 , please review and merge my PR. And also please create a new nuget release.

File: Plugin.NetStandardStorage.Abstractions/Interfaces/IFolder.cs [hunk 1]
```diff
@@ -90,5 +90,11 @@ public interface IFolder
         /// Deletes the current folder.
         /// </summary>
         void Delete();
+
+        /// <summary>
+        /// Deletes the current folder with all of its content.
+        /// Warning: Recurse delete.
+        /// </summary>
+        void DeleteWithContent();
```
Comment (laurentiustamate94) [inline Plugin.NetStandardStorage.Abstractions/Interfaces/IFolder.cs]: please change to `DeleteRecursively`

File: Plugin.NetStandardStorage/Implementations/Folder.cs [hunk 2]
```diff
@@ -236,8 +236,9 @@ public IList<IFolder> GetFolders()
                 .AsReadOnly();
         }
 
-        public bool CheckFileExists(string path)
+        public bool CheckFileExists(string fileName)
         {
+            var path = Path.Combine(FullPath, fileName);
             return System.IO.File.Exists(path);
```
Comment (laurentiustamate94) [inline Plugin.NetStandardStorage/Implementations/Folder.cs]: new line here, before the return. I want initialization to be separated vertically from other concepts, like return in this case

File: Plugin.NetStandardStorage/Implementations/Folder.cs [hunk 3]
```diff
@@ -258,5 +259,18 @@ public void Delete()
...
+        public void DeleteWithContent()
+        {
+            if (!this._canDelete)
+            {
+                throw new IOException("The root folder can't be deleted !");
+            }
+
+            if (this.CheckFolderExists(this.FullPath))
+            {
+                Directory.Delete(this.FullPath, true);
```
Comment (laurentiustamate94) [inline Plugin.NetStandardStorage/Implementations/Folder.cs]: please specify what `true` means by specifying the `recursive` parameter name, like this: `Directory.Delete(this.FullPath, recursive: true);`

**Review by laurentiustamate94 — CHANGES_REQUESTED** [review]
please do make the changes and after this I will merge gladly

**Review by laurentiustamate94 — APPROVED** [review]

**Pull request merged**